src/main/java/com/yurjinia/common/security/jwt/filters/JwtAuthenticationFilter.java

```diff
@@ -28,36 +32,81 @@ public class JwtAuthenticationFilter extends OncePerRequestFilter {
     @Override
     protected void doFilterInternal(@NonNull HttpServletRequest request,
                                     @NonNull HttpServletResponse response,
-                                    @NonNull FilterChain filterChain) throws ServletException, IOException {
-        final String authHeader = request.getHeader(JwtConstants.AUTHORIZATION_HEADER);
+                                    @NonNull FilterChain filterChain) throws IOException {
+        try {
+            String path = request.getRequestURI();
+            if (isAuthUrl(path)) {
+                filterChain.doFilter(request, response);
+                return;
+            }
+
+            String token = getJwt(request);
+            validateTokenAndSetAuthentication(token, request);
 
-        if (authHeader == null || !authHeader.startsWith(JwtConstants.TOKEN_PREFIX)) {
             filterChain.doFilter(request, response);
-            return;
+        } catch (CommonException ex) {
+            handleCommonException(response, ex);
+
+        } catch (Exception ex) {
+            handleException(response, ex);
         }
+    }
 
-        final String jwt = authHeader.substring(JwtConstants.TOKEN_PREFIX.length());
-        final String userEmail = jwtService.extractUsername(jwt);
+    private boolean isAuthUrl(String path) {
+        return path.startsWith(JwtConstants.AUTH_URL);
+    }
+
+    private void validateTokenAndSetAuthentication(String token, HttpServletRequest request) throws CommonException {
```

Comment:
too many responsibilities in one method. violation of SRP principle of SOLID.

try to do it similar way as below:

private void validateTokenAndSetAuthentication(String token, HttpServletRequest request) throws CommonException {
    if (!StringUtils.hasText(token) || !JwtValidator.isValidateFormat(token)) {
        throw new JwtInvalidException("Invalid JWT format");
    }

    String userEmail = jwtService.extractUsername(token).orElseThrow(() -> new JwtInvalidException("No username found in JWT"));
    if (isAlreadyAuthenticated()) {
        return;
    }

    UserDetails userDetails = userDetailsService.loadUserByUsername(userEmail);
    validateAndAuthenticateToken(token, userDetails, request);
}

private boolean isAlreadyAuthenticated() {
    return SecurityContextHolder.getContext().getAuthentication() != null;
}

private void validateAndAuthenticateToken(String token, UserDetails userDetails, HttpServletRequest request) throws CommonException {
    if (!jwtService.isTokenValid(token, userDetails)) {
        throw new JwtInvalidException("JWT token is not valid");
    }
    setAuthentication(userDetails, request);
}

also put setAuthentication outside of validateAndAuthenticateMethod.

Reply:
I'm not sure if I need to write down exactly what the problem is. Since this is authorisation, it is enough for the user to know that the invalid token